app/services/agent_removal.rb

```diff
@@ -25,5 +25,20 @@ def upcoming_rdvs?
   def should_soft_delete?
     (@agent.organisations - [@organisation]).empty?
   end
+
+  def errors
+    I18n.t("admin.territories.agent_roles.destroy.cannot_delete_because_of_rdvs") if upcoming_rdvs?
+  end
+
+  def confirm
```

Comment:
On pourrait avoir un nom de méthode un peu plus descriptif ici
```suggestion
  def confirmation_message
```